### PR TITLE
Ignore deleted users for CheckEmailIsUnique and CheckUserExists

### PR DESCRIPTION
**Pull request by Rynoxx**
So after reworking the old WHMCS module I’ve noticed that the behaviour of CheckUserExists method in the manage_clients module and the ctrl_users::CheckEmailIsUnique is, in my opinion, somewhat unexpected.

They do count deleted accounts towards the lack of uniqueness in usernames and email addresses while the MySQL queries for username (GitHub Search for “ac_user_vc”) and email (GitHub Search for “ac_email_vc”) are always (at least the instances that I could find) accompanied by userid or a check that ac_deleted_ts IS NULL, with the exception being queries regarding the zadmin user of the panel.

Seeing as this causes issues when trying to add users manually or via the API, usernames and emails of accounts which are “deleted” can still not be used by new ones and the “deleted” account would have to manually be deleted from the MySQL table. I’m curious as to why and/or if this is the intended behaviour of the functions.

Bottom line, what are your thoughts on changing the previously mentioned methods to only count the accounts which haven’t been deleted towards the lack of uniqueness of a username/email?

(Also took the liberty of making the email check guaranteed to be case insensitive)

### CODE REVIEW

**Comment by TGates71** [comment]
This is built by design to prevent using the same email for multiple accounts (which also breaks the lost password functions) Also, it stores all deleted info for your records and it dis-allows the re-use of already entered information *domains, dns, users, etc.) to help prevent SPAM users from registering over and over again if their account has been deleted.

Although, if you checked our forums you would have found the 'Deleted Records Manager' module which allows the zadmin to delete these stored records for testing/allowing a user to re-apply with same information. Bout your PR, I'll leave that to the devs ;) @MBlagui

**Comment by Rynoxx** [comment]
Thanks TGates!
That explains it, with good reasoning indeed.  
In regards to the forgot password functionality, at a glance it seems like it should work, since it looks for accounts which haven't been deleted and multiple accounts under the same email shouldn't be possible with my approach, except for deleted ones ofc (which judging by the search results for "ac_email_vc" in the github repo shouldn't be an issue for the queries that I could find).  

I didn't think about the Deleted Records Manager when I wrote this, although I have seen it when browsing the forums before, I'll keep it in mind and recommend it to people if they find that this is an issue and they haven't found it themselves. :)

**Comment by TGates71** [comment]
I'm not sure if this PR will be incorporated. As I mentioned it is done on purpose. Users are suggested to use the Deleted Records Manager to manually remove any 'old' entries. Unless, a check is created for the zadmin to allow/disallow the use of previously used email addresses. (I think this was being looked into and/or saved for a later release.) This could easily be added as a DB or manual setting in the configs.
```php
$allow_old_emails == '0';  // Allow = 1 / Disallow = 0 (Default = 0)
if ($allow_old_email == '1') {
    $sql = "SELECT COUNT(*) FROM x_accounts WHERE LOWER(ac_email_vc)=LOWER(:email) AND ac_deleted_ts IS NULL"; 
} else {
    $sql = "SELECT COUNT(*) FROM x_accounts WHERE LOWER(ac_email_vc)=:email";
}
```
@MBlagui Your thoughts?

**Comment by Rynoxx** [comment]
I just realized that this exact thing was mentioned in #155 (which was closed) so this PR might as well be disregarded, and I'll resort back to using custom functions for my use-case, unless the devs have changed their mind. (Which seems best, after rethinking this)
